File: desktop/src/app/module/core/domain/persona.ts
import {User} from './user';
import {Documento} from './documento';
import {Domicilio} from './domicilio';
import {Turno} from './turno';
import {Contacto} from './contacto';
import {Plan} from './plan';

export class Persona {
  id?: number;
  apellido?: string;
  fechaNacimiento?: number;
  genero?: string;
  nombre?: string;
  userAsociado?: User;
  integrantes?: Persona[] = [];
  documento?: Documento;
  estadoCivil?: string;
  domicilio?: Domicilio;
  contacto?: Contacto[] = [];
  turno?: Turno[] = [];
  plan?: Plan;
}

File: microservices/commons/src/main/java/net/folderit/domain/Persona.java
package net.folderit.domain;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import net.folderit.domain.enums.EstadoCivil;
import net.folderit.domain.enums.Genero;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;
import java.util.List;

import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;

@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(NON_NULL)
public class Persona implements Serializable {
    private Long id;
    private String apellido;
    private Date fechaNacimiento;
    private Genero genero;
    private String nombre;
    private boolean enabled = Boolean.TRUE;
    private User userAsociado;
    private Documento documento;
    private EstadoCivil estadoCivil;
    private Domicilio domicilio;
    private List<Persona> integrantes;
    private List<Contacto> contacto;
    private List<Turno> turno;
}